src/main/java/Spike.java

```diff
@@ -0,0 +1,135 @@
+            } else if (input.startsWith("mark")) {
+                int index = Integer.parseInt(input.substring(5)) - 1;
+                inputList[index].setDone(true);
+                displayMarkMsg(index, inputList);
+            } else if (input.startsWith("unmark")) {
+                int index = Integer.parseInt(input.substring(7)) - 1;
+                inputList[index].setDone(false);
+                displayUnmarkMsg(index, inputList);
+            } else if (input.startsWith("todo")) {
+                inputList[iter] = new Todo(processTodo(input));
+                displayAcknowledgement(inputList, iter);
+            } else if (input.startsWith("deadline")) {
+                inputList[iter] = new Deadline(processDeadline(input));
+                displayAcknowledgement(inputList, iter);
+            } else if (input.startsWith("event")) {
+                inputList[iter] = new Event(processEvent(input));
+                displayAcknowledgement(inputList, iter);
+            } else if (input.contentEquals("bye")) {
+                break;
+            } else {
+                System.out.println("Not valid");
+                iter -= 1;
+            }
+            iter += 1;
+        }
+        displayByeMsg();
+    }
+
+    private static void displayAcknowledgement(Task[] inputList, int iter) {
+        char Badge = getBadge(inputList[iter]);
+        System.out.println(DIVIDER);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(" [" + Badge + "]" + "[ ] " + inputList[iter].description);
+        System.out.println("Now you have " + (iter + 1) + " tasks in the list.");
+        System.out.println(DIVIDER);
+    }
+
+    private static String processEvent(String input) {
+        String event = input.substring(6);
+        String[] parts = event.split(" /from ");
+        String[] time = parts[1].split(" /to ");
+        return parts[0] + " (from: " + time[0] + " to: " + time[1] + ")";
+    }
+
+    private static String processDeadline(String input) {
+        String deadline = input.substring(9);
+        String[] parts = deadline.split(" /by ");
+        return parts[0] + " (by: " + parts[1] + ")";
+    }
+
+    private static String processTodo(String input) {
+        return input.substring(5);
+    }
+
+    private static void displayAddedMsg(String input) {
+        System.out.println(DIVIDER);
+        System.out.println("added: " + input);
+        System.out.println(DIVIDER);
+    }
+
+    private static void displayList(Task[] inputList) {
+        System.out.println(DIVIDER);
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < inputList.length; i++) {
+            Task value = inputList[i];
+            if (value == null) {
+                break;
+            }
+            char Badge = getBadge(value);
+            System.out.println((i + 1) + ".[" + Badge + "]" + "[" + value.getStatusIcon() + "] " + value.description);
```

Comment:
Avoid long lines of code
Try to add line breaks to reduce the length of code on each line (<110 - 120 characters, ideally 80 characters)